app/src/main/java/ch/epfl/sdp/drone3d/map/offline/OfflineMapSaverImpl.kt

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.map.offline
+
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.mapbox.mapboxsdk.camera.CameraPosition
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.geometry.LatLngBounds
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.offline.*
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import timber.log.Timber
+
+
+/**
+ * Used internally to facilitate the serialization of the OfflineRegionMetadata data class
+ */
+@Serializable
+private data class SerializableRegionMetadata(val name:String, val latSouth:Double, val lonWest:Double,
+                                              val latNorth:Double, val lonEast:Double,val tileCount:Long, val zoom:Double,
+                                              val isDownloading:Boolean)
+
+class OfflineMapSaverImpl(val context:Context, private val styleUri:String):OfflineMapSaver {
+
+    companion object{
+        private const val JSON_CHARSET = "UTF-8"
+        const val MAX_ZOOM = 20.0
+        const val MAX_ZOOM_DISCREPANCY = 6.0 //Maximum difference between max zoom and min zoom
+        const val TILE_LIMIT = 6000L //6000 tiles corresponds to Greater London with zoom 0-15
```

Comment:
Could MAX_ZOOM, MAX_ZOOM_DISCREPANCY and TILE_LIMIT be private ?

Reply:
Yes, it is done